### PR TITLE
AV-220811: Remove code related with empty hostname at gateway listener and httproute

### PR DESCRIPTION
**Pull request by akshayhavile**
…

### CODE REVIEW

**Comment by ashishnayak96** [comment]
build ako

**Review by ashishnayak96 — APPROVED** [review]
LGTM

**Review by arihantg — APPROVED** [review]
LGTM. Please attach Gateway FT Results

File: tests/gatewayapitests/status/httproute_test.go [hunk 1]
```diff
@@ -79,23 +71,21 @@ func TestHTTPRouteWithValidConfig(t *testing.T) {
...
 			Type:    string(gatewayv1.RouteConditionAccepted),
 			Reason:  string(gatewayv1.RouteReasonAccepted),
 			Status:  metav1.ConditionTrue,
 			Message: "Parent reference is valid",
-		}, {
-			Type:   string(gatewayv1.RouteConditionResolvedRefs),
-			Reason: string(gatewayv1.RouteReasonResolvedRefs),
-			Status: metav1.ConditionTrue,
-		}}
+		}
```
Comment (pkoshtavmware) [inline tests/gatewayapitests/status/httproute_test.go:87]: Why is this condition removed ?

Comment (akshayhavile) [reply tests/gatewayapitests/status/httproute_test.go]: It was merge issue.

**Comment by ashishnayak96** [comment]
build ako

File: tests/gatewayapitests/status/httproute_test.go [hunk 2]
```diff
@@ -1113,8 +1115,6 @@ func TestHTTPRouteWithValidAndInvalidBackendKind(t *testing.T) {
 	svcName := "avisvc-hr-17"
 
 	akogatewayapitests.SetupGatewayClass(t, gatewayClassName, akogatewayapilib.GatewayController)
-	integrationtest.CreateSVC(t, DEFAULT_NAMESPACE, svcName, "TCP", corev1.ServiceTypeClusterIP, false)
```
Comment (pkoshtavmware) [inline tests/gatewayapitests/status/httproute_test.go:1116]: Is this also a merge issue or was it purposely removed ?

Comment (pkoshtavmware) [reply tests/gatewayapitests/status/httproute_test.go:1116]: Kindly revert this change

**Comment by akshayhavile** [comment]
build ako

**Review by pkoshtavmware — APPROVED** [review]
LGTM

**Review by arihantg — APPROVED** [review]
LGTM

**Comment by akshayhavile** [comment]
build ako

**Comment by ashishnayak96** [comment]
build ako

**Comment by akshayhavile** [comment]
build ako

**Comment by akshayhavile** [comment]
Pasting result:
```+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+
| Absolute Path:Test case                                                          |     Results    |       Duration      |  Start Time |   End Time  |
+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+
|      Test Suite Path and Name : avitest/functional/ako/ako-gw-                   |                |                     |             |             |
| api/test_http_route.py                                                           |                |                     |             |             |
| ++++++++++++++++++++++++++++++++++++++++                                         |                |                     |             |             |
| TestHttpRoute.test_setup_configuration                                           |   [1;32m  PASSED  [0m   |       00:03:06      |   12:55:08  |   12:58:15  |
| TestHttpRoute.test_insecure_httproute                                            |   [1;32m  PASSED  [0m   |       00:01:05      |   12:58:15  |   12:59:20  |
| TestHttpRoute.test_secure_httproute                                              |   [1;32m  PASSED  [0m   |       00:01:04      |   12:59:20  |   13:00:24  |
| TestHttpRoute.test_httproute_match_crud                                          |   [1;32m  PASSED  [0m   |       00:01:52      |   13:00:24  |   13:02:17  |
| TestHttpRoute.test_httproute_match_append_header_crud                            |   [1;32m  PASSED  [0m   |       00:01:52      |   13:02:17  |   13:04:10  |
| TestHttpRoute.test_httproute_match_replace_header_crud                           |   [1;32m  PASSED  [0m   |       00:01:52      |   13:04:10  |   13:06:03  |
| TestHttpRoute.test_httproute_backendref_crud                                     |   [1;32m  PASSED  [0m   |       00:01:35      |   13:06:03  |   13:07:38  |
| TestHttpRoute.test_httproute_multiple_rules                                      |   [1;32m  PASSED  [0m   |       00:01:38      |   13:07:38  |   13:09:17  |
| TestHttpRoute.test_http_route_with_multiple_gateways                             |   [1;32m  PASSED  [0m   |       00:02:08      |   13:09:17  |   13:11:25  |
| TestHttpRoute.test_multiple_http_routes_with_gateway                             |   [1;32m  PASSED  [0m   |       00:02:39      |   13:11:25  |   13:14:04  |
| TestHttpRoute.test_http_route_same_path_different_headers                        |   [1;32m  PASSED  [0m   |       00:01:38      |   13:14:04  |   13:15:42  |
| TestHttpRoute.test_httproute_multiple_hosts                                      |   [1;32m  PASSED  [0m   |       00:02:07      |   13:15:42  |   13:17:50  |
| TestHttpRoute.test_http_route_wild_crd_fqdn                                      |   [1;32m  PASSED  [0m   |       00:01:05      |   13:17:50  |   13:18:56  |
| TestHttpRoute.test_http_route_no_fqdn                                            |   [1;32m  PASSED  [0m   |       00:01:05      |   13:18:56  |   13:20:01  |
| TestHttpRoute.test_traffic_split                                                 |   [1;32m  PASSED  [0m   |       00:03:53      |   13:20:01  |   13:23:55  |
| TestHttpRoute.test_httproute_default_backend                                     |   [1;32m  PASSED  [0m   |       00:02:11      |   13:23:55  |   13:26:06  |
| TestHttpRoute.test_httproute_with_no_match                                       |   [1;31m  PASSED  [0m   |       00:02:25      |   13:26:06  |   13:28:32  |
| TestHttpRoute.test_delete_configurations                                         |   [1;32m  PASSED  [0m   |       00:01:11      |   13:28:33  |   13:29:45  |
| -------------------------------------------------------------------------------- |  ------------  |     -----------     | ----------- | ----------- |
|                                                                                  | [1;32m  Passed: 17 [0m  |                     |             |             |
|                                                                                  |    [1;31mFailed:1[0m    |                     |             |             |
|  Total test cases: 18                                                            |    [1;33mSkipped:0[0m   | Total Time:00:34:34 |             |             |
+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+```

**Pull request merged**